atc/creds/vault/vault_test.go

```diff
@@ -274,3 +297,687 @@ var _ = Describe("Vault", func() {
 		})
 	})
 })
+
+// The below tests use ghttp handlers to mock a real vault API to the api_client.
+// AppendHandlers has the following behavior which make the tests a bit messy.
+
+//The first incoming request is handled by the first handler in the list, the
+//second by the second, etc...
+
+// As such, if a url is requested out of order or not for the proper path, the
+// test fails with a corresponding error.
+// Its annoying, but it works (if theres a better solution feel free to refactor this)
+
+var _ = Describe("Vault KV2", func() {
+
+	var server *ghttp.Server
+	var returnedMountInfo vaultapi.Secret
+	var statusCodeOK int
+
+	var v *vault.Vault
+	var variables vars.Variables
+	var vaultApi *vault.APIClient
+	var varFoo vars.Reference
+
+	BeforeEach(func() {
+		server = ghttp.NewServer()
+
+		var err error
+		vaultApi, err = vault.NewAPIClient(lagertest.NewTestLogger("test"), "http://"+server.Addr(), vault.TLSConfig{}, vault.AuthConfig{}, "")
```

Comment:
to catch up with `master`:

```suggestion
		vaultApi, err = vault.NewAPIClient(lagertest.NewTestLogger("test"), "http://"+server.Addr(), vault.TLSConfig{}, vault.AuthConfig{}, "", 0)
```